sdcm/cluster_aws.py

```diff
@@ -402,37 +390,47 @@ def add_nodes(self, count, ec2_user_data='', dc_idx=0, enable_auto_bootstrap=Fal
         return added_nodes
 
     def node_setup(self, node, verbose=False, timeout=3600):
-        if not cluster.Setup.REUSE_CLUSTER:
-            node.wait_ssh_up(verbose=verbose)
-            authenticator = self.params.get('authenticator')
-            endpoint_snitch = self.params.get('endpoint_snitch')
+        endpoint_snitch = self.params.get('endpoint_snitch')
+        def node_config_setup():
             if len(self.datacenter) > 1:
-                if not endpoint_snitch:
-                    endpoint_snitch = "Ec2MultiRegionSnitch"
-                node.datacenter_setup(self.datacenter)
                 node.config_setup(seed_address=self.get_seed_nodes_by_flag(private_ip=False),
                                   enable_exp=self._param_enabled('experimental'),
                                   endpoint_snitch=endpoint_snitch,
                                   broadcast=node.public_ip_address,
-                                  authenticator=authenticator,
+                                  authenticator=self.params.get('authenticator'),
                                   server_encrypt=self._param_enabled('server_encrypt'),
                                   client_encrypt=self._param_enabled('client_encrypt'),
                                   append_scylla_args=self.params.get('append_scylla_args'))
             else:
                 node.config_setup(enable_exp=self._param_enabled('experimental'),
                                   endpoint_snitch=endpoint_snitch,
-                                  authenticator=authenticator,
+                                  authenticator=self.params.get('authenticator'),
                                   server_encrypt=self._param_enabled('server_encrypt'),
                                   client_encrypt=self._param_enabled('client_encrypt'),
                                   append_scylla_args=self.params.get('append_scylla_args'))
 
+        if not cluster.Setup.REUSE_CLUSTER:
+
+            node.wait_ssh_up(verbose=verbose)
+            if len(self.datacenter) > 1:
+                if not endpoint_snitch:
+                    endpoint_snitch = "Ec2MultiRegionSnitch"
+                node.datacenter_setup(self.datacenter)
+            node_config_setup()
+
             node.remoter.run('sudo yum install -y {}-gdb'.format(node.scylla_pkg()),
                              verbose=verbose, ignore_status=True)
             node.remoter.run('sudo systemctl restart scylla-server.service')
 
         node.wait_db_up(verbose=verbose, timeout=timeout)
         node.remoter.run('nodetool status', verbose=True, retry=5)
 
+        if node.replacement_node_ip:
```

Comment:
This is copy-paste from cluster.py:BaseNode.node_setup().
As a software engineers we should avoid such practice. 
Please:
- generalize `def node_config_setup():` method and add it to BaseNode base class
- create clear_replacement_node_config() that will call node config setup